frontend/deb/templates/custom_enable_postinst.tmpl

```diff
@@ -0,0 +1,17 @@
+# Automatically added by github.com/Azure/dalec
+# This is generally copied from dh_installsystemd
+# This is required because dh_installsystemd operates on the unit base name and
+# not the full unit name (e.g. "foo" instead of "foo.service") and sometimes
+# we want to enable one and not the other.
+if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
+	# This will only remove masks created by d-s-h on package removal.
+	deb-systemd-helper unmask '{{.}}' >/dev/null || true
+
+	if deb-systemd-helper --quiet was-enabled '{{.}}'; then
```

Comment:
How does this work if this is the first time the unit is being installed? Will the unit still be enabled?

Reply:
This is all verbatim from the deb-systemd-helper scripts.
Running this command locally on an non-existant package:

```console
$ DPKG_MAINTSCRIPT_PACKAGE=1 deb-systemd-helper was-enabled totally-does-not-exist; echo $?                                                                                                                                          ─╯
enabled
0
```

Reply:
Reading the code for all this (and enabling debug on the script) it is reading some state in `/var/lib/systemd/deb-systemd-helper-enabled/`.
This directory has files for each unit that d-s-h manages.
Each file contains 1 or more references to systemd units linked into whatever target its being installed to (e.g. multi-user.target).
The `was-enabled` subcommand checks if all the listed links are present (and are symlinks).
If there is no file (or the file is not a symlink) it says yes it is enabled.

Reply:
Ok perfect, that was exactly my question: I was concerned about what the `was-enabled` command would return in the case where d-s-h had no record of a unit. But since this is handled properly then this seems fine to me (especially since it mimics the behavior of the deb-systemd-helper scripts anyways)